api/envoy/api/v2/auth/cert.proto

```diff
@@ -238,12 +238,29 @@ message CommonTlsContext {
   // Configs for fetching TLS certificates via SDS API.
   repeated SdsSecretConfig tls_certificate_sds_secret_configs = 6;
 
+  message CombinedCertificateValidationContext {
+    // How to validate peer certificates.
+    CertificateValidationContext default_validation_context = 1;
+
+    // Config for fetching validation context via SDS API.
+    SdsSecretConfig validation_context_sds_secret_config = 2;
+  };
+
   oneof validation_context_type {
     // How to validate peer certificates.
     CertificateValidationContext validation_context = 3;
 
     // Config for fetching validation context via SDS API.
     SdsSecretConfig validation_context_sds_secret_config = 7;
+
+    // Combined certificate validation context holds a default CertificateValidationContext
+    // and SDS config. When SDS server returns dynamic CertificateValidationContext, both dynamic
+    // and default CertificateValidationContext are merged into a new CertificateValidationContext
+    // for validation. This merge is done by Message::MergeFrom(), so dynamic
+    // CertificateValidationContext overwrites singular fields in default
+    // CertificateValidationContext, and concatenates repeated fields to default
+    // CertificateValidationContext.
```

Comment:
I think we need some extra check during init for this (and document here) that until both default and overriding cvc are received from control plane, the cluster / listener will reject connections. Empty trusted ca means no validation, so it is important to make sure both are received.

Reply:
I think with the latest change, we don't need extra check. ContextConfigImpl::isReady() tells us if secret config is ready. Only when received CVC and default CVC can merge into a valid CVC, the cluster / listener will use the secret to handle connections.